index.js

```diff
@@ -37,9 +37,13 @@ function compose (middleware) {
     function dispatch (i) {
       if (i <= index) return Promise.reject(new Error('next() called multiple times'))
       index = i
-      const fn = middleware[i] || next
-      if (!fn) return Promise.resolve()
+      let fn = middleware[i]
       try {
+        if (fn && context.wrappers) {
+          context.wrappers.forEach((wrapper) => { fn = wrapper(fn) })
```

Comment:
Not that it matters, but I find `wrapper => { ... }` without ()'s around single argument fat-arrows.